### PR TITLE
add chunk alloc system variable

### PR DESCRIPTION
**Pull request by Oreoxmt**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/11853
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by tiancaiamao — APPROVED** [review]

File: system-variables.md [hunk 1]
```diff
@@ -321,6 +321,12 @@ This variable is an alias for [`last_insert_id`](#last_insert_id).
 - Default value: `OFF`
 - This variable is used to show whether the execution plan used in the previous `execute` statement is taken directly from the plan cache.
 
+### `last_sql_use_alloc` <span class="version-mark">New in v6.4.0</span>
+
+- Scope: SESSION
+- Default value: `OFF`
+- This variable is read-only. It is used to show whether the previous statement uses a cached chunk object (Chunk allocation).
```
Comment (shichun-0415) [inline system-variables.md:328]:
```suggestion
- This variable is read-only. It is used to show whether the previous statement uses a cached chunk object (chunk allocation).
```

File: system-variables.md [hunk 2]
```diff
@@ -2372,6 +2386,22 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
 - The default value is `CANCEL`, but in TiDB v4.0.2 and earlier versions, the default value is `LOG`.
 - This setting was previously a `tidb.toml` option (`oom-action`), but changed to a system variable starting from TiDB v6.1.0.
 
+### `tidb_max_reuse_chunk` <span class="version-mark">New in v6.4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `64`
+- Range: `[0, 2147483647]`
+- This variable controls the maximum cached chunk objects of chunk allocation. Adjusting this variable to a larger value might increase the risk of OOM.
```
Comment (shichun-0415) [inline system-variables.md:2395]:
```suggestion
- This variable controls the maximum cached chunk objects of chunk allocation. Adjusting this variable to too large a value might increase the risk of OOM.
```

File: system-variables.md [hunk 3]
```diff
@@ -2372,6 +2386,22 @@ For a system upgraded to v5.0 from an earlier version, if you have not modified
...
+
+### `tidb_max_reuse_column` <span class="version-mark">New in v6.4.0</span>
+
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
+- Default value: `256`
+- Range: `[0, 2147483647]`
+- This variable controls the maximum cached column objects of chunk allocation. Adjusting this variable to a larger value might increase the risk of OOM.
```
Comment (shichun-0415) [inline system-variables.md:2403]:
```suggestion
- This variable controls the maximum cached column objects of chunk allocation. Adjusting this variable to too large a value might increase the risk of OOM.
```

**Review by shichun-0415 — APPROVED** [review]
LGTM

File: system-variables.md [hunk 4]
```diff
@@ -1493,6 +1499,14 @@ MPP is a distributed computing framework provided by the TiFlash engine, which a
 - This variable is used to control whether to enable TiDB mutation checker, which is a tool used to check consistency between data and indexes during the execution of DML statements. If the checker returns an error for a statement, TiDB rolls back the execution of the statement. Enabling this variable causes a slight increase in CPU usage. For more information, see [Troubleshoot Inconsistency Between Data and Indexes](/troubleshoot-data-inconsistency-errors.md).
 - For new clusters of v6.0.0 or later versions, the default value is `ON`. For existing clusters that upgrade from versions earlier than v6.0.0, the default value is `OFF`.
 
+### tidb_enable_reuse_chunk <span class="version-mark">New in v6.4.0</span>
```
Comment (shichun-0415) [inline system-variables.md:1502]: According to the source PR, this variable should be moved to Line 1662.

**Review by shichun-0415 — APPROVED** [review]

File: tidb-configuration-file.md [hunk 5]
```diff
@@ -17,6 +17,20 @@ The TiDB configuration file supports more options than command-line parameters.
 - Default value: `true`
 - It is recommended to set it to `false` if you need to create a large number of tables (for example, more than 100 thousand tables).
 
+### `tidb_max_reuse_chunk` <span class="version-mark">New in v6.4.0</span>
+
+- Controls the maximum cached chunk objects of chunk allocation. Adjusting this variable to too large a value might increase the risk of OOM.
```
Comment (shichun-0415) [inline tidb-configuration-file.md:22]:
```suggestion
- Controls the maximum cached chunk objects of chunk allocation. Setting this configuration item to too large a value might increase the risk of OOM.
```

File: tidb-configuration-file.md [hunk 6]
```diff
@@ -17,6 +17,20 @@ The TiDB configuration file supports more options than command-line parameters.
...
+- Default value: `64`
+- Minimum value: `0`
+- Maximum value: `2147483647`
+
+### `tidb_max_reuse_column` <span class="version-mark">New in v6.4.0</span>
+
+- Controls the maximum cached column objects of chunk allocation. Adjusting this variable to too large a value might increase the risk of OOM.
```
Comment (shichun-0415) [inline tidb-configuration-file.md:29]:
```suggestion
- Controls the maximum cached column objects of chunk allocation. Setting this configuration item to too large a value might increase the risk of OOM.
```

**Pull request merged**